common/src/main/java/com/wynntils/core/functions/arguments/parser/ArgumentParser.java

```diff
@@ -16,7 +16,17 @@ public final class ArgumentParser {
     public static ErrorOr<FunctionArguments> parseArguments(
             FunctionArguments.Builder argumentsBuilder, String rawArgs) {
         if (rawArgs == null || rawArgs.isEmpty()) {
-            return ErrorOr.of(argumentsBuilder.buildWithDefaults());
+            // 1, If there are no arguments, return early.
+            if (argumentsBuilder.getArgumentCount() == 0) {
+                return argumentsBuilder.buildWithValues(List.of());
+            }
+
+            // 2, If there are required arguments, return an error, otherwise return the default arguments.
+            if (argumentsBuilder instanceof FunctionArguments.OptionalArgumentBuilder optionalArgumentBuilder) {
+                return ErrorOr.of(optionalArgumentBuilder.buildWithDefaults());
+            } else {
+                return ErrorOr.error("Missing required arguments: (%s)".formatted(argumentsBuilder.getArgumentNames()));
```

Comment:
This assumes its some kind of RequiredArgumentBuilder we got then..? I probably missed the full understanding of this in the previous PRs.

Reply:
Builder is required, OptionalRequirementBuilder is not

Reply:
(it is not on per argument basis, rather in function basis)